tests/framework/extensions/workloads/pods/pod_status.go

```diff
@@ -78,3 +66,36 @@ func StatusPods(client *rancher.Client, clusterID string) ([]string, []error) {
 
 	return podResults, podErrors
 }
+
+// CheckPodStatus is a helper function that uses the steve client to check the status of a single pod
+func CheckPodStatus(pod *v1.SteveAPIObject) (podResults string, podError error, err error) {
+	podStatus := &corev1.PodStatus{}
+	err = v1.ConvertToK8sType(pod.Status, podStatus)
+	if err != nil {
+		return "", nil, err
+	}
+	if podStatus.ContainerStatuses == nil || len(podStatus.ContainerStatuses) == 0 {
+
+		return fmt.Sprintf("WARN: %s: Container Status is Empty \n", pod.Name), nil, nil
+	}
+
+	image := podStatus.ContainerStatuses[0].Image
+	phase := podStatus.Phase
+
+	if phase == corev1.PodFailed || phase == corev1.PodUnknown {
+		logrus.Infof("Pod %s: Not active | Image %s", pod.Name, image)
+		// Do not report as error if pod is in failed state due to container termination
+		for _, containerStatus := range podStatus.ContainerStatuses {
+			if containerStatus.State.Terminated == nil {
+				return "", fmt.Errorf("ERROR: %s: %s", pod.Name, podStatus), nil
```

Comment:
Small nit here. There are cases where a helm operation pod might be terminated but another spawned and was successful. This might cause polling to be exited too soon.

Reply:
discussed offline, will leave as-is until we find a better solution